.buildkite/pipeline.yml

```diff
@@ -28,6 +28,8 @@ steps:
         cargo-deny --all-features check bans
       fi
       RUSTFLAGS='-D warnings' cargo check --workspace --all-targets --all-features
+      RUSTFLAGS='-D warnings' cargo check -p neard
+      RUSTFLAGS='-D warnings' cargo check -p neard --features nightly_protocol,nightly_protocol_features
```

Comment:
This feels repetitive to me. I think we don't have to run all these checks together. cc @matklad

Reply:
Syntactically, `RUSTFLAGS='-D warnings'` is repetitive. It'd be cool to set it once globally for all jobs, not sure if buildkite syntax allows for that. 

Semantically, the two extra jobs are not repetitive. It's totally possible that `cargo check --workspace` works, but `cargo check -p neard` doesn't, due to how features are activated in Cargo. The backstory here is that we did broke release build, which we didn't notice because CI was only testing workspace mode. 

Cost-wise, I'd expect the two additional checks to be realtively cheap -- they should be able to re-use artifacts for all of the dependencies.